### PR TITLE
removed dependency on bart tokenizer(slow/fast) LED

### PR DESCRIPTION
**Pull request by Infrared1029**
# What does this PR do?

removes the dependency between LED tokenizer and Bart's (slow version)

Fixes # (issue)

follows hugging's face philosophy of Do Repeat Yourself

## Who can review?
@sgugger

### CODE REVIEW

**Review by Infrared1029 — COMMENTED** [review]
~The docstring where it shows how to use the tokenizer, what LED model goes there? (ive put a comment there asking)~, nvm found it

File: src/transformers/models/led/tokenization_led.py [hunk 1]
```diff
@@ -40,18 +49,378 @@
...
+    """
+    pairs = set()
+    prev_char = word[0]
+    for char in word[1:]:
+        pairs.add((prev_char, char))
+        prev_char = char
+    return pairs
+
+
+# Copied from transformers.models.bart.tokenization_bart.BartTokenizer with Bart->LED
```
Comment (sgugger) [inline src/transformers/models/led/tokenization_led.py:93]:
```suggestion
# Copied from transformers.models.bart.tokenization_bart.BartTokenizer with Bart->LED,facebook/bart-base->allenai/led-base-16384
```
We also need to update the checkpoint name here.

File: src/transformers/models/led/tokenization_led.py [hunk 2]
```diff
@@ -40,18 +49,378 @@
...
+            Whether or not to add an initial space to the input. This allows to treat the leading word just as any
+            other word. (BART tokenizer detect beginning of words by the preceding space).
+    """
+
+    vocab_files_names = VOCAB_FILES_NAMES
     pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
     max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
+    model_input_names = ["input_ids", "attention_mask"]
+
+    # Copied from transformers.models.bart.tokenization_bart.BartTokenizer.__init__
```
Comment (sgugger) [inline src/transformers/models/led/tokenization_led.py:175]:
```suggestion
```
You have the global copied from on the class, so you won't need this one.

File: src/transformers/models/led/tokenization_led.py [hunk 3]
```diff
@@ -40,18 +49,378 @@
...
+    @property
+    # Copied from transformers.models.bart.tokenization_bart.BartTokenizer.vocab_size
+    def vocab_size(self):
+        return len(self.encoder)
+
+    # Copied from transformers.models.bart.tokenization_bart.BartTokenizer.get_vocab
+    def get_vocab(self):
+        return dict(self.encoder, **self.added_tokens_encoder)
+
+    # Copied from transformers.models.bart.tokenization_bart.BartTokenizer.bpe
```
Comment (sgugger) [inline src/transformers/models/led/tokenization_led.py:239]: Same for those and all methods below.

File: src/transformers/models/led/tokenization_led_fast.py [hunk 4]
```diff
@@ -41,19 +47,251 @@
 }
 
 
-class LEDTokenizerFast(BartTokenizerFast):
+# Copied from transformers.models.bart.tokenization_bart_fast.BartTokenizerFast with Bart->LED
```
Comment (sgugger) [inline src/transformers/models/led/tokenization_led_fast.py:50]:
```suggestion
# Copied from transformers.models.bart.tokenization_bart_fast.BartTokenizerFast with Bart->LED,facebook/bart-base->allenai/led-base-1638
```

File: src/transformers/models/led/tokenization_led_fast.py [hunk 5]
```diff
@@ -41,19 +47,251 @@
...
+            Whether the post processing step should trim offsets to avoid including whitespaces.
     """
 
+    vocab_files_names = VOCAB_FILES_NAMES
     pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
     max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
     slow_tokenizer_class = LEDTokenizer
+    model_input_names = ["input_ids", "attention_mask"]
+
+    # Copied from transformers.models.bart.tokenization_bart_fast.BartTokenizerFast.__init__
```
Comment (sgugger) [inline src/transformers/models/led/tokenization_led_fast.py:136]: And same here: no need for other Copied from statements.

**Review by sgugger — COMMENTED** [review]
Thanks a lot for your PR! I've left a couple of comments to fix some of your Copied from statements.

With your rebase, it looks like the PR contains modifications that should not be there. I think the easiest fix for that is to close this one and open a fresh one from the same branch :-)

**Comment by Infrared1029** [comment]
Thanks a lot for the feedback! ill fix everything and open a brand new one.

**Pull request closed without merging**